src/azure-cli-core/azure/cli/core/commands/arm.py

```diff
@@ -339,16 +340,20 @@ def _get_child(parent, collection_name, item_name, collection_key):
         return result
 
 
-def _get_operations_tmpl(cmd):
-    operations_tmpl = cmd.command_kwargs.get('operations_tmpl',
-                                             cmd.command_kwargs.get('command_type').settings['operations_tmpl'])
+def _get_operations_tmpl(cmd, custom_command=False):
+    if custom_command:
+        operations_tmpl = cmd.command_kwargs.get('operations_tmpl') or \
+            cmd.command_kwargs.get('custom_command_type').settings['operations_tmpl']
+    else:
+        operations_tmpl = cmd.command_kwargs.get('operations_tmpl') or \
+            cmd.command_kwargs.get('command_type').settings['operations_tmpl']
     if not operations_tmpl:
         raise CLIError("command authoring error: cmd '{}' does not have an operations_tmpl.".format(cmd.name))
     return operations_tmpl
 
 
-def _get_client_factory(_, kwargs):
-    command_type = kwargs.get('command_type', None)
+def _get_client_factory(_, custom_command=False, **kwargs):
+    command_type = kwargs.get('custom_command_type', None) if custom_command else kwargs.get('command_type', None)
```

Comment:
It seems like rather than having lots of logic like this:
```Python
kwargs.get('custom_command_type', None) if custom_command else kwargs.get('command_type', None)
```

Where you risk a logic error if there's a typo, you could just leverage the `source_kwarg` argument you removed elsewhere. That way you specify it in fewer places and your logic looks like:

```Python
kwargs.get(source_kwarg, None)
```

You also wouldn't need that big if-else block in `_get_operations_tmpl` because you would just pass it the `source_kwarg` directly.